internal/provider/utils.go

```diff
@@ -37,6 +38,22 @@ func GetOk[T any](d *schema.ResourceData, key string) (T, bool, error) {
 	return t, true, fmt.Errorf("%w: %s to %T: %v", errTypecast, key, t, raw)
 }
 
+func parsePrivateKey(d *schema.ResourceData, privateKey string) (ssh.Signer, error) {
+	privateKeyPass, ok, err := GetOk[string](d, "conn.0.private_key_pass")
+	if ok {
+		if err != nil {
+			return nil, err
+		}
+		return ssh.ParsePrivateKeyWithPassphrase([]byte(privateKey), []byte(privateKeyPass))
+	}
+	if strings.Contains(privateKey, "Proc-Type: 4,ENCRYPTED") {
+		if privateKeyPass == "" {
+			return nil, fmt.Errorf("rsa private key is encrypted, password required but not supplied")
+		}
+	}
```

Comment:
```suggestion
```

Given that not all password protected private keys will have `Proc-Type: 4,ENCRYPTED`, I think the default `ssh: this private key is passphrase protected` error message will be sufficient here.

Reply:
The user sould know if they missed the password. Just give a confusing error, is a bad design.

(i saw this check in other projects)